### ISSUE TITLE
utility pallet is missing MaxBatchLimit configuration

### ISSUE THREAD

**Issue by GopherJ** [paritytech/substrate#10658]
currently utility.batch_all has no limit on the number of extrinsics. The parachains normally have only 0.5s to prepare a block. Image we have a big utility.batch_all, the parachain can stop producing blocks forever and keep warning:

```
Block production took too long
```

**Comment by GopherJ** [comment]
This may not only applies to `utility.batch_all` but also all big blocks, is there a way to skip such big blocks?

**Comment by bingryan** [comment]
good issue.i did an experiment on this.also have some doubts about this.

first, when I send 100 tasks use `Utility batch`, the parachain is normally ok.
but when this increased to 1000 tasks, the parachain stopped. and get error:

```
aura: [Parachain] ⌛️ Discarding proposal for slot 136845516; block production took too long
```

**Comment by bkchr** [comment]
> Hackers can use this to attack parachains

If you don't properly benchmark your chain, this is true. However, there is no need for a `MaxBatchLimit`, because you have a weight assigned to each call and that is respected by the batch weight calculation. If you can construct such a call that the weight says it fits into the block, but the block takes too long to be produced, it means your weights are incorrect.

**Comment by KiChjang** [comment]
> does that mean if weight correctly calculated, such huge extrinsic will just fail during some check phase without chain stuck?

Yes, the [`CheckWeight` signed extension](https://github.com/paritytech/substrate/blob/9eddaec3b8f15eec0c458e7bb3a1c16db91d8199/frame/system/src/extensions/check_weight.rs#L45-L66) will always check whether the extrinsic weight exceeds the configured individual dispatch class extrinsic weight, **and** whether it exceeded the remaining weights available for the block before the extrinsic is applied.

The fact that this issue exists is very concerning, and should not happen under properly configured chains. Did some of you not benchmark your extrinsics in a way that attempts to hit the worst-case complexity for your dispatchable functions? Or did some of you created a runtime that omitted the `CheckWeight` signed extension? Or did some of you misconfigure your maximum weights for each dispatch class?

In any case, as far as I can tell, this isn't an issue that requires code changes on Substrate, unless there are any outstanding issues I missed?
